core/src/test/scala/spinal/core/sim/SpinalFunSuite.scala

```diff
@@ -10,6 +10,99 @@ class Identity extends Component {
   io.output := io.input
 }
 
+class BadIdentity extends Identity {
+  io.output := io.output
+}
+
+
+// Drawback: I have to provide a base type as I do not take arguments anymore
+class IdentityFlatSpec extends SpinalFlatSpec[Identity] {
+
+  // Global parameters
+  override val config = SimConfig.withWave
+  // override val caching = false
+
+  // Parameters can be erased by local parameter when building benches
+  val asIs = bench(new Identity, caching = false)
+  // I can transform, and the "software abstraction" type is inferred
+  val transformed = benchTransform(new Identity, dut => SoftIdentity(dut))
+  // These ones will not compile
+  val failing = bench(new BadIdentity)
+  val failingNoCache = bench(new BadIdentity, caching = false)
+
+  // Before my first test, it tests if it compiles, and cancels test if it doesn't
+  // Here I provide an empty test and I get:
+  //   failing should compile *** FAILED ***
+  //   failing should do nothing !!! CANCELED !!!
+  failing should "do nothing" in { dut => }
+
+  //    failingNoCache should do nothing *** FAILED ***
+  failingNoCache should "do nothing" in { dut => }
+
+  // Notice that benches are `Nameable` (but I did not implement recursive names)
```

Comment:
So in `val theName = bench(XXX)` I get `theName` in the test names later, see below.